controllers/goflowkube/goflowkube_objects.go

```diff
@@ -24,25 +23,53 @@ const configPath = "/etc/goflow-kube"
 const configFile = "config.yaml"
 
 const (
-	healthServiceName       = "health"
-	healthTimeoutSeconds    = 5
-	livenessPeriodSeconds   = 10
-	startupFailureThreshold = 5
-	startupPeriodSeconds    = 10
+	healthServiceName = "health"
+
+/* 	healthTimeoutSeconds    = 5
+livenessPeriodSeconds   = 10
+startupFailureThreshold = 5
+startupPeriodSeconds    = 10 */
 )
 
 // PodConfigurationDigest is an annotation name to facilitate pod restart after
 // any external configuration change
 const PodConfigurationDigest = "flows.netobserv.io/goflow-kube-config"
 
-type ConfigMap struct {
-	Listen      string        `json:"listen,omitempty"`
-	Loki        LokiConfigMap `json:"loki,omitempty"`
-	PrintInput  bool          `json:"printInput"`
-	PrintOutput bool          `json:"printOutput"`
+type Config struct {
+	LogLevel string   `json:"log-level,omitempty"`
+	Pipeline Pipeline `json:"pipeline,omitempty"`
 }
 
-type LokiConfigMap struct {
+type Pipeline struct {
+	Ingest    Ingest      `json:"ingest"`
+	Decode    Decode      `json:"decode"`
+	Encode    Encode      `json:"encode"`
+	Extract   Extract     `json:"extract"`
+	Transform []Transform `json:"transform"`
+	Write     Write       `json:"write"`
+}
+type Ingest struct {
+	Collector Collector `json:"collector"`
+	Type      string    `json:"type"`
+}
+type Collector struct {
```

Comment:
@jotak this works ... It is possible to use the struct from here:  https://github.com/netobserv/flowlogs2metrics/blob/051a608b1368f25986bcfb1880b416e7d57d943a/pkg/api/ingest_collector.go#L20 directly instead of reimplementing the API struct. The value is that we can be sure that we are aligned to some version of flowlogs2metrics.

Reply:
Yes, but until now we've tried to keep the operator loosely coupled with other models to avoid dependency complexity and keep vendors light. We can revisit this choice if we feel it's useful.